qiskit/algorithms/phase_estimators/hamiltonian_phase_estimation.py

```diff
@@ -128,9 +148,9 @@ def _get_unitary(self, hamiltonian, pe_scale, evolution) -> QuantumCircuit:
     def estimate(
         self,
         hamiltonian: OperatorBase,
-        state_preparation: Optional[StateFn] = None,
-        evolution: Optional[EvolutionBase] = None,
-        bound: Optional[float] = None,
+        state_preparation: StateFn | None = None,
+        evolution: EvolutionBase | None = None,
```

Comment:
Could we move to supporting the opflow-free* variants here, too? So

* hamiltonian -> `BaseOperator | PauliSumOp`
* state_preparation -> `QuantumCircuit | Statevector`: The state preparation can convert a statevector to a circuit by using `QuantumCircuit.prepare_state`.
* evolution -> `EvolutionSynthesis | None`: If the operator is a matrix we can use the `HamiltonianGate`, otherwise we can use the `PauliEvolutionGate` with a specified synthesis method from `qiskit.synthesis.evolution` 

The order and name of the arguments could stay the same, just the type would have to be updated 🙂 

*) modulo `PauliSumOp` which we still have supported in the primitives right now

Reply:
Since these are in-situ updates the existing function has to continue to work but we should be able to check instance types so we can deprecate their use favoring the new types you indicate which we can add.

Reply:
Yeah, sorry I should've been less lazy in typing. By "move to" I meant "start supporting the new, and pending-deprecate the old" 😉